### PR TITLE
New assets page: categories

### PR DESCRIPTION
**Pull request by jonator**
## What is the purpose of the change

Screenshot 2024-03-18 at 3 12 32 PM

This adds asset categories filter to the `getAssets` function. It also adds the basic categories UI to the new assets table. 

I also did some refactoring to remove async behavior from the `getAsset` and `getAssets` functions. I updated our error handling to adapt to this, since the functions throw when an asset is not found. The reason I decided to remove it was that we we're doing unnecessary caching with `cachified` around getting assets, and processing the in-memory asset list was found to be cheap computationally; basically caching the processing of the asset list was a pre-optimization.

Also, I removed the consequential use of async wherever they are used, as well as allowing the pools providers to filter pools with unlisted assets without capturing those errors in Sentry.

### ClickUp Task

[ClickUp Task URL](https://app.clickup.com/t/86a0vf6v6)

## Brief Changelog

* Add categories filter to `getAssets` and add to tRPC procedures
* Remove async from assets functions, and update uses throughout codebase to match this. Also changed how errors are captured to assist with sync functions.
* Add category filter UI
* I code gen the list of available categories into our config/generated folder. I extracted generation of TS files into a util, and updated the generate-asset-lists script to use that util.
* Prevent unnecessary re renders of assets table rows while scrolling (by using a useCallback for the functions passed to table props)

## Testing and Verifying

Categories filter works. The pools providers still work and tests pass.

## Summary by CodeRabbit

- **New Features**
	- Enhanced asset management with category selection, filtering, and improved asset information display.
- **Refactor**
	- Streamlined asset and pool queries for better efficiency and readability.
- **Tests**
	- Adjusted test cases to align with changes in function calls and error handling.

### CODE REVIEW

File: packages/web/utils/error.ts [hunk 1]
```diff
@@ -1,7 +1,24 @@
...
+}
+
+/** Captures an error from a throwable function and returns the value or `undefined` if it throws. */
+export function captureIfError<TReturn>(throwableFn: () => TReturn) {
+  try {
+    return throwableFn();
+  } catch (e) {
+    captureError(e);
+  }
+}
```
Comment (JoseRFelix) [inline packages/web/utils/error.ts:24]: Thoughts on adding unit tests for these critical functions?

File: packages/web/localizations/__tests__/localizations.test.ts [hunk 2]
```diff
@@ -9,6 +9,9 @@ import { glob } from "glob";
 // use to silence warnings
 const warn = (..._args: Parameters<typeof console.warn>) => null; // console.warn(...args);
 
+/** Add key paths here to skip them in localizations tests. */
+const omittedKeyPaths = ["assets.categories"];
```
Comment (JoseRFelix) [inline packages/web/localizations/__tests__/localizations.test.ts:13]: Can you give me more context on why this key is being omitted?

**Review by JoseRFelix — COMMENTED** [review]
Great work! Left some comments

File: packages/web/localizations/__tests__/localizations.test.ts (hunk 2 above)
Comment (jonator) [reply packages/web/localizations/__tests__/localizations.test.ts:13]: I dynamically append the generated category ID to `assets.categories` translation key, so I do `"assets.categories" + category` in the code

**Review by JoseRFelix — APPROVED** [review]

**Pull request merged**